theseus/src/state/mod.rs

```diff
@@ -4,17 +4,18 @@ use std::sync::Arc;
 use tokio::sync::{Mutex, OnceCell, RwLock, Semaphore};
 
 // Submodules
-mod dirs;
-pub use self::dirs::*;
-
-mod metadata;
-pub use metadata::*;
-
-mod settings;
-pub use settings::*;
+macro_rules! submodule {
+    ($name:ident) => {
+        mod $name;
+        pub use self::$name::*;
+    };
+}
 
-mod profiles;
-pub use profiles::*;
+submodule!(dirs);
+submodule!(metadata);
+submodule!(profiles);
+submodule!(settings);
+submodule!(users);
```

Comment:
Could we change this to normal declaration? It's not much code save, and I think the other deceleration is more explicit (also macros like this don't really work that well with IntelliJ Rust for some reason)

Reply:
Alright, will do

Reply:
If there were a better solution for async, the authors of the Tokio runtime probably would have used it as the main way to trace errors in their own runtime (after all, Tokio is the one who authored the tracing library and is using it in their console debugger).